soroban-sdk/src/crypto.rs

```diff
@@ -3,11 +3,19 @@
 use crate::{
     env::internal::{self, BytesObject},
     unwrap::UnwrapInfallible,
-    Bytes, BytesN, ConversionError, Env, IntoVal, TryFromVal, Val,
+    Bytes, BytesN, ConversionError, Env, IntoVal, Symbol, TryFromVal, TryIntoVal, Val, Vec, U256,
 };
 
 pub mod bls12_381;
 pub mod bn254;
+pub(crate) mod poseidon2_params;
+pub mod poseidon2_sponge;
```

Comment:
Can we move the sponge and params inside a `poseidon2` module? Same with `poseidon`?